crates/router/tests/connectors/payeezy.rs

```diff
@@ -0,0 +1,761 @@
+#[actix_web::test]
+async fn should_fail_void_payment_for_auto_capture() {
+    let authorize_response = CONNECTOR
+        .authorize_payment(
+            PayeezyTest::get_payment_data(),
+            PayeezyTest::get_payment_info(),
+        )
+        .await
+        .expect("Authorize payment response");
+    let txn_id = utils::get_connector_transaction_id(authorize_response.response.clone()).unwrap();
+    let capture_connector_meta = utils::get_connector_meta(authorize_response.response);
+    let capture_response = CONNECTOR
+        .capture_payment(
+            txn_id,
+            Some(types::PaymentsCaptureData {
+                connector_meta: capture_connector_meta,
+                ..utils::PaymentCaptureType::default().0
+            }),
+            PayeezyTest::get_payment_info(),
+        )
+        .await
+        .expect("Capture payment response");
+    assert_eq!(capture_response.status, enums::AttemptStatus::Charged);
+    let txn_id = utils::get_connector_transaction_id(capture_response.clone().response);
+    let connector_meta = utils::get_connector_meta(capture_response.response);
+    assert_ne!(txn_id, None, "Empty connector transaction id");
+    let void_response = CONNECTOR
+        .void_payment(
+            txn_id.unwrap(),
+            Some(types::PaymentsCancelData {
+                connector_transaction_id: String::from(""),
+                cancellation_reason: Some("requested_by_customer".to_string()),
+                connector_meta,
+                ..Default::default()
+            }),
+            PayeezyTest::get_payment_info(),
+        )
+        .await
+        .expect("Void payment response");
+    assert_eq!(
+        void_response.response.unwrap_err().message,
+        "Bad Request (26) - Invalid Amount"
+    );
+}
+
+// Captures a payment using invalid connector payment id.
+#[actix_web::test]
+async fn should_fail_capture_for_invalid_payment() {
+    let response = PayeezyTest {}
+        .authorize_payment(PayeezyTest::get_payment_data(), None)
+        .await
+        .unwrap();
+    assert_eq!(response.status, enums::AttemptStatus::Authorized);
+    let connector_payment_id = "123455678".to_string();
+    let connector_meta = utils::get_connector_meta(response.response);
+    let capture_data = types::PaymentsCaptureData {
+        connector_meta,
+        amount_to_capture: Some(50),
+        ..utils::PaymentCaptureType::default().0
+    };
+    let capture_response = PayeezyTest {}
+        .capture_payment(connector_payment_id, Some(capture_data), None)
+        .await
+        .unwrap();
+    assert_eq!(
+        capture_response.response.unwrap_err().message,
+        String::from("Bad Request (69) - Invalid Transaction Tag")
+    );
+}
+
+// Refunds a payment with refund amount higher than payment amount.
+#[actix_web::test]
+async fn should_fail_for_refund_amount_higher_than_payment_amount() {
+    let authorize_response = CONNECTOR
+        .authorize_payment(
```

Comment:
This is to test refund amount > payment amount scenario. So you can use `CONNECTOR.make_payment` to reduce the length of the code written for this test.